src/components/CookieWarning.js

```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash/fp'
+import { useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { ButtonPrimary, ButtonSecondary, Link } from 'src/components/common'
+import { Ajax } from 'src/libs/ajax'
+import { signOut } from 'src/libs/auth'
+import colors from 'src/libs/colors'
+import * as Nav from 'src/libs/nav'
+import { getLocalPref, setLocalPref } from 'src/libs/prefs'
+import { authStore } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+
+
+export const cookiesAcceptedKey = 'cookiesAccepted'
+
+export const CookieWarning = () => {
+  const signal = Utils.useCancellation()
+  const [acceptedCookies, setAcceptedCookies] = useState(getLocalPref(cookiesAcceptedKey))
+  authStore.subscribe(state => {
+    setAcceptedCookies(getLocalPref(cookiesAcceptedKey))
+  })
+  const acceptCookies = acceptedCookies => {
+    setLocalPref(cookiesAcceptedKey, acceptedCookies)
+    setAcceptedCookies(getLocalPref(cookiesAcceptedKey))
+  }
+  const rejectCookies = async () => {
+    const cookies = document.cookie.split(';')
+    acceptCookies(false)
+    await Ajax(signal).Runtimes.invalidateCookie()
+    // Expire all cookies
+    _.forEach(cookie => {
+      // Find an equals sign and uses it to grab the substring of the cookie that is its name
+      const eqPos = cookie.indexOf('=')
+      const cookieName = eqPos > -1 ? cookie.substr(0, eqPos) : cookie
+      document.cookie = `${cookieName}=;expires=Thu, 01 Jan 1970 00:00:00 GMT`
+    }, cookies)
```

Comment:
do we do this because we can't delete the user's cookies, so we expire them instead?

Reply:
Yes, this is the standard way to delete cookies, and then the browser will clean them up. There are libraries which abstract this, but we do very little cookie management, so its probably fine not to add a library and to do it by hand.